tiledb/sm/query_plan/query_plan.h

```diff
@@ -65,8 +65,9 @@ class QueryPlan {
    * Constructor
    *
    * @param query A query object for which we want to calculate the plan
+   * @param strategy The strategy the query, required only for remote queries
    */
-  QueryPlan(Query& query);
+  QueryPlan(Query& query, const std::string strategy = "");
```

Comment:
Do we ever use the default parameter for strategy? If not we should remove it.

Reply:
Yes, we use it for local queries, which is the default case. For remote queries the deserialization function will pass the strategy received from `capnp` in order to create a QueryPlan, as the strategy needs to come from the server.